contracts/token/ERC721/ERC721.sol

```diff
@@ -396,8 +406,8 @@ abstract contract ERC721 is Context, ERC165, IERC721, IERC721Metadata, IERC721Er
      * Emits an {ApprovalForAll} event.
      */
     function _setApprovalForAll(address owner, address operator, bool approved) internal virtual {
-        if (owner == operator) {
-            revert ERC721InvalidOperator(owner);
+        if (operator == owner || operator == address(0)) {
```

Comment:
Same as in my other comment I don't really know why we would disallow `operator == owner`. I think we need to remove that check.